src/base-element.js

```diff
@@ -589,40 +591,101 @@ export class BaseElement {
+   * @param {boolean=} opt_isDefault Whether this action is the default.
    * @public
    */
-  registerAction(method, handler, minTrust = ActionTrust.HIGH) {
+  registerAction(
+    method, handler, minTrust = ActionTrust.HIGH, opt_isDefault = false) {
     this.initActionMap_();
-    this.actionMap_[method] = {handler, minTrust};
+    dev().assert(() => {
+      if (opt_isDefault) {
+        for (const i in this.actionMap_) {
+          const action = this.actionMap_[i];
+          if (action.isDefault) {
+            return false;
+          }
+        }
+      }
+      return true;
+    }, 'There is already a default action registered.');
+    this.actionMap_[method] = {handler, minTrust, isDefault: opt_isDefault};
+  }
+
+  /**
+   * Registers the default action for this component.
+   * @param {function(!./service/action-impl.ActionInvocation)} handler
+   * @param {string=} methodName
+   */
+  registerDefaultAction(handler, methodName = DEFAULT_METHOD_) {
+    this.registerAction(
+        methodName, handler, ActionTrust.HIGH, /* opt_isDefault */ true);
   }
 
   /**
    * Requests the element to execute the specified method. If method must have
    * been previously registered using {@link registerAction}, otherwise an
    * error is thrown.
    * @param {!./service/action-impl.ActionInvocation} invocation The invocation data.
-   * @param {boolean} unusedDeferred Whether the invocation has had to wait any time
+   * @param {?Array<{tagOrTarget: !Element, method: string}>} opt_whitelist
+   *     Actions permitted for execution.
+   * @param {boolean=} unusedDeferred Whether the invocation has had to wait any time
    *   for the element to be resolved, upgraded and built.
    * @final
    * @package
    */
-  executeAction(invocation, unusedDeferred) {
-    if (invocation.method == 'activate') {
+  executeAction(invocation, opt_whitelist = [], unusedDeferred) {
+    const whitelist = opt_whitelist || [];
+    if (invocation.method == DEFAULT_METHOD_) {
       if (invocation.satisfiesTrust(this.activationTrust())) {
-        return this.activate(invocation);
+        for (const actionName in this.actionMap_) {
+          const action = this.actionMap_[actionName];
+          // Check if the action is whitelisted. Note that we pass in
+          // the actual method/action name and not 'activate', which is the
+          // default name/method, so that we don't have to whitelist
+          // 'activate', as it is implied.
+          const actionWhitelisted = this.isActionWhitelisted_(
+              invocation, actionName, whitelist);
+          const {method, tagOrTarget} = invocation;
+          if (!actionWhitelisted) {
+            user().error(TAG_,
+                `"${tagOrTarget}.${method}" is not whitelisted ` +
+                `whitelist: ${JSON.stringify(whitelist)}.`);
+          }
+          if (action.isDefault && actionWhitelisted) {
+            return action.handler(invocation);
+          }
+        }
```

Comment:
This seems overly complex. Instead of storing an `isDefault` bit on every entry in `actionMap_`, can we just store a single string as the alias for the default action? E.g. `this.defaultActionAlias_`. 

There should only be one (e.g. "open" for amp-sidebar) and we can just do:
```js
if (this.defaultActionAlias_) {
  const whitelisted = this.isActionWhitelisted_(invocation, this.defaultActionAlias_, whitelist);
}
```